### ISSUE TITLE
Uploading attachment with different extensions in filename and stream's content

### ISSUE THREAD

**Issue by BuzkoYaroslav** [XeroAPI/xero-node#363]
**My version of the xero-node is 3.1.2**

**Desciption of a problem**
My team and I have encountered a bug with sending attachments through uploadAttachmentFromStream method. When we were trying to upload attachment from stream and attachment content type was pdf but in filename we have .png or .jpeg extension it resulted first of all in long export and lastly that in Xero we saw infinite spinner with our attachment eventually not uploaded.

**Steps to reproduce the behavior:**
1. Use method uploadAttachmentFromStream
2. As an argument into method above pass object with filename with extension .png, stream that contains data of the pdf file and mimeType application/pdf.
3. Go to Xero organization you exported the document.
4. See infinite spinner at the place an attachment should be.

**Expected behavior**
Filename can be whatever you pass maybe except special characters and there should be a dependency only between mimeType and stream content.

**Comment by SerKnight** [comment]
Thank you for submitting this bug @BuzkoYaroslav - I will try to reproduce this in the next working session with attachments.. 

You might want to reference a sample app upload example, as we do require passing the content type in the header.. I have yet to implement the `stream` functions but will do my best to get those examples in soon

https://github.com/XeroAPI/xero-node-oauth2-app/blob/master/src/app.ts#L995

**Comment by RettBehrens** [comment]
A quick note on why we’re closing this issue:

With the release of version 4.x of the Xero-Node SDK in December of 2019, we delivered a complete overhaul of the SDK. 

We moved from writing our SDKs by hand to generating our SDKs using our [Xero-OpenAPI](https://github.com/XeroAPI/Xero-OpenAPI) specs. By utilizing Xero-OpenAPI specs to generate code we’ve improved the developer experience of maintaining the SDK and improved the accessibility for our developer community to contribute via pull requests to the Xero-OpenAPI specs. 

We also built our new SDKs to use the industry standard OAuth2 in preparation for our March 2021 deprecation of OAuth1. For more details, see our [blog post on our decision to move to OAuth2](https://devblog.xero.com/an-update-on-why-we-are-saying-goodbye-oauth-1-0a-hello-oauth-2-0-6a839230908f). 

Version 3.x using Oauth1 and version 4.x using OAuth2 do not share any code as we now generate our SDKS. Going forward we’re focusing on building, maintaining and improving our generated SDKs but you are welcome to fork the [‘oauth1’ branch](https://github.com/XeroAPI/xero-node/tree/oauth1) and apply changes if you wish to implement something for your integration.
